string map_pgm_request
---
string map_pgm_str
